front/app/containers/ProjectsShowPage/timeline/index.tsx

```diff
@@ -111,50 +109,31 @@ const ProjectTimelineContainer = memo<Props & WithRouterProps>(
 
     useEffect(() => {
       if (
-        !isNilOrError(location) &&
+        selectedPhase !== null &&
         !isNilOrError(phases) &&
-        phases.length > 0
+        project &&
+        !isNilOrError(locale)
       ) {
-        const currentPhase = getCurrentPhase(phases);
-        const firstPhase = getFirstPhase(phases);
-        const lastPhase = getLastPhase(phases);
-        const lastPastPhase = getLastPastPhase(phases);
-
-        // if, coming from the siteMap, a phase url parameter was passed in, we pick that phase as the default phase,
-        // then remove the param so that when the user navigates to other phases there is no mismatch
-        if (isString(location?.query?.phase)) {
-          const phase = phases.find(
-            (phase) => phase.id === location.query.phase
-          );
-
-          if (phase) {
-            window.history.replaceState(null, '', location.pathname);
-            selectPhase(phase);
-          }
-        } else if (currentPhase) {
-          selectPhase(currentPhase);
-        } else if (
-          firstPhase &&
-          pastPresentOrFuture([
-            firstPhase.attributes.start_at,
-            firstPhase.attributes.end_at,
-          ]) === 'future'
-        ) {
-          selectPhase(firstPhase);
-        } else if (
-          lastPastPhase &&
-          lastPhase &&
-          pastPresentOrFuture([
-            lastPhase.attributes.start_at,
-            lastPhase.attributes.end_at,
-          ]) === 'future'
-        ) {
-          selectPhase(lastPastPhase);
+        setPhaseURL(selectedPhase.id, phases, project, locale);
+      }
+    }, [selectedPhase]);
```

Comment:
You also have dependecies on project, phases and locale here.

Reply:
True, but I don't really need to update the component if any of those things change, right? The only thing this hook has to respond to is the `selectedPhase`

Reply:
It can take a while before these are loaded. So if there are first undefined, but after that have a different value, we still want to call the useEffect, no?

Reply:
Yes but it is impossible for `selectPhase` to exist before `phases` is loaded, since it is... a phase. And `phases` will only be loaded when the project is available. I don't know about `locale`, maybe I should still add that one?

Reply:
My knowledge of the streams don't go that far. selectedPhase is coming from a different stream than phases, so (without knowing for sure) I think it's possible one is there before the other.

Likewise, project won't necessarily be there before phases because we use the projectId we're getting from the props. Phases doesn't depend on project being there in this component. 

locale is one you also need to add, yep, because if someone changes the locale while using the platform, the phase URL needs to be adjusted.

So my train of thought is a bit: when I'm not 100% sure and I don't have the time to figure it out, I rather add a few more dependencies than cause bugs. :-)

Reply:
Fair enough :+1: